addons/point_of_sale/static/src/js/Screens/OrderManagementScreen/ControlButtons/InvoiceButton.js

```diff
@@ -0,0 +1,159 @@
+odoo.define('point_of_sale.InvoiceButton', function (require) {
+    'use strict';
+
+    const { useListener } = require('web.custom_hooks');
+    const { useContext } = owl.hooks;
+    const { isRpcError } = require('point_of_sale.utils');
+    const PosComponent = require('point_of_sale.PosComponent');
+    const OrderManagementScreen = require('point_of_sale.OrderManagementScreen');
+    const OrderFetcher = require('point_of_sale.OrderFetcher');
+    const Registries = require('point_of_sale.Registries');
+    const contexts = require('point_of_sale.PosContext');
+
+    class InvoiceButton extends PosComponent {
+        constructor() {
+            super(...arguments);
+            useListener('click', this._onClick);
+            this.orderManagementContext = useContext(contexts.orderManagement);
+        }
+        get selectedOrder() {
+            return this.orderManagementContext.selectedOrder;
+        }
+        set selectedOrder(value) {
+            this.orderManagementContext.selectedOrder = value;
+        }
+        get isAlreadyInvoiced() {
+            if (!this.selectedOrder) return false;
+            return Boolean(this.selectedOrder.account_move);
+        }
+        get commandName() {
+            if (!this.selectedOrder) {
+                return 'Invoice';
+            } else {
+                if (this.selectedOrder.isFromClosedSession) return 'Cannot Invoice';
```

Comment:
Small error with the condition here, for all orders from previous sessions already invoiced, it is called "Cannot Invoice", and should be "Reprint Invoice", (The button works, so if you click on cannot invoice it'll reprint it).

you probably need something like: this.selectedOrder.isFromClosedSession and not this.isAlreadyInvoiced

Reply:
You are right. It should show Reprint Invoice if the order from closed session is already invoiced.